SimTKcommon/src/Pathname.cpp

```diff
@@ -238,6 +240,161 @@ void Pathname::deconstructPathname( const string&   name,
     }
 }
 
+void Pathname::deconstructPathnameUsingSpecifiedWorkingDirectory(const std::string& swd,
+                                                                 const std::string& path,
+                                                                 std::string& directory,
+                                                                 std::string& fileName,
+                                                                 std::string& extension)
+{
+    directory.erase(); fileName.erase(); extension.erase();
+    string pathdrive, swddrive, finaldrive;
+    String processed;
+
+    // Remove all the white space and make all the slashes be forward ones.
+    // (For Windows they'll be changed to backslashes later.)
+    String pathcleaned = String::trimWhiteSpace(path).replaceAllChar('\\', '/');
+    if (pathcleaned.empty())
+        return; // path consisted only of white space
```

Comment:
Do you think this is the right policy for an empty pathname? Would it make more sense to return with directory=processed swd, fileName=extension=""? If it should be changed you could do it now or we could just file an issue and deal with it later.